libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -366,10 +376,20 @@ static void eventCallback( MQTTContext_t * pContext,
     assert( pContext != NULL );
     assert( pPacketInfo != NULL );
 
+    if( pPacketInfo->type == disconnectOnPacketType )
+    {
+        /* Terminate MQTT connection with server for session restoration test. */
+        TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+
+        /* Terminate TLS session and TCP connection to test session restoration
+         * across network connection. */
+        ( void ) Openssl_Disconnect( &networkContext );
+    }
+
     /* Handle incoming publish. The lower 4 bits of the publish packet
      * type is used for the dup, QoS, and retain flags. Hence masking
      * out the lower bits to check if the packet is publish. */
-    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    else if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
```

Comment:
`else if` with no braces makes introducing bugs during maintenance very easy so we should avoid it.  This is a pretty long function anyway, so you might consider splitting it into two: one to handle lines 387-417 and another for lines 421-500.

Reply:
Broken down the `eventCallback` function